cordapp-cpk2/src/main/java/net/corda/plugins/cpb2/CpbTask.java

```diff
@@ -73,6 +78,20 @@ private boolean isCPK(@NotNull FileTreeElement element) {
         final Path cpkPath = element.getFile().toPath();
         try (JarInputStream cpkStream = new JarInputStream(new BufferedInputStream(Files.newInputStream(cpkPath)))) {
             final String cpkType = cpkStream.getManifest().getMainAttributes().getValue(CORDA_CPK_TYPE);
+            final String cpkCordappName = cpkStream.getManifest().getMainAttributes().getValue(CPK_CORDAPP_NAME);
+            System.out.println("Processing CPK " + cpkType + " " + cpkCordappName);
+            if (cpkCordappName != null) {
+                System.out.println(cpkCordappName);
+                if (cpkNames.containsKey(cpkCordappName)) {
+                    if (cpkNames.get(cpkCordappName) == 1) {
+                        cpkNames.put(cpkCordappName, 2);
```

Comment:
So you're allowing two cpks of the same name? I assume this is one for workflow and one for contract? Is it possible to make this check a bit more foolproof?

Reply:
The plugin seems to iterate through each CPK twice or for some other reason calls this function twice on each CPK, so one instance of the name appearing will be counted twice

Reply:
You should be able to call this function as many times as you like. Shouldn't `cpkNames` be local to this method?

Maybe it would help get to the bottom of this behaviour if you logged all the cpkNames? I don't think we can merge this without knowing what's going on.